### DISCUSSION TITLE
Idea: Toggling between multiple code versions.

### DISCUSSION THREAD

**Discussion by PlastoJesta** [jupyter/notebook#7290] [Ideas]
I would like a feature similar to ChatGPT, with which I can switch between different code blocks and name them. For example, I have a code block and name it Version1, then I create a new code block which I name Version2, containing a revised version of the code block. By switching the code blocks, the respective one is then executed. This allows for quickly testing different scenarios by setting a different config for each code block, other variables, etc. It is also possible to find an older version of the code at an appropriate place.

**Comment by jtpio** [comment]
@PlastoJesta maybe Jupyter AI could already help you with this? https://github.com/jupyterlab/jupyter-ai

Otherwise maybe it could be worth asking there.
